app/components/avo/views/resource_index_component.rb

```diff
@@ -136,7 +136,8 @@ def show_search_input
   def authorized_to_search?
     # Hide the search if the authorization prevents it
     return true unless resource.authorization.respond_to?(:has_action_method?)
-    return false unless resource.authorization.has_action_method?("search")
+    return true unless resource.authorization.has_action_method?("search")
+    return false unless resource.authorization.has_action_method?("avo_search?")
```

Comment:
`has_action_method?(argument)` will pick the `argument` and use it as key to access the `Avo.configuration.authorization_methods` hash that looks like:

```ruby
      @authorization_methods = {
        index: "index?",
        show: "show?",
        edit: "edit?",
        new: "new?",
        update: "update?",
        create: "create?",
        destroy: "destroy?"
      }
```

`avo_search?` is something specific that the user can set like:

```ruby
Avo.configure do |config|
  # ...

  ## == Authorization ==
  config.authorization_methods = {
    # ...
    search: "avo_search?" # override this method
  }

  # ...
end
```

It can be even `can_search_or_not?` if the user configure it with:

```ruby
Avo.configure do |config|
  # ...

  ## == Authorization ==
  config.authorization_methods = {
    # ...
    search: "can_search_or_not?" # override this method
  }

  # ...
end
```

We should always access the `search` key to verify if the method exists.